cassandra-analytics-integration-tests/src/test/java/org/apache/cassandra/analytics/correctness/BulkWriteCorruptionTest.java

```diff
@@ -53,22 +56,55 @@
 import static org.apache.cassandra.testing.TestUtils.ROW_COUNT;
 import static org.apache.cassandra.testing.TestUtils.TEST_KEYSPACE;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.fail;
 
-public class BulkWriteDiskCorruptionTest extends SharedClusterSparkIntegrationTestBase
+public class BulkWriteCorruptionTest extends SharedClusterSparkIntegrationTestBase
 {
-    private static final QualifiedName QUALIFIED_NAME = new QualifiedName(TEST_KEYSPACE, "test_write_disk_corruption");
+    enum CorruptionMode
+    {
+        DISK,
+        WIRE
+    }
+
+    private static final QualifiedName QUALIFIED_NAME = new QualifiedName(TEST_KEYSPACE, "test_write_corruption");
 
     static
     {
-        // Intercepts SortedSSTableWriter#validateSSTables to corrupt file on purpose.
         // Install the class rebase the earliest, before JVM loads the class
-        BBHelperInterceptSortedSSTableWriterValidateSSTables.install();
+        BBHelperFileCorrupter.install();
```

Comment:
NIT: Why not just have 2 different byte buddy helper classes, rather than introducing the mode enum/if statements in two different intercepted methods?

Reply:
Yep. It sounds better.

Reply:
Made the change and I realized why having the switch. 

Each test case only expect corruption at specific locations. Without the switch, `testDataTransferCorruption` has the files corrupted on disk and fail.

Reply:
I think there may be a way around this, but it seems like it requires some more knowledge of ByteBuddy, and I don't have that knowledge right now, so I don't want to hold up this PR.